docs/kb/semgrep-ci/github-upload-findings-in-security-dashboard.md

```diff
@@ -35,35 +35,44 @@ Changing the repository's default workflow permissions changes the permissions f
 This job only requires `write` permissions for `security-events`.
 
 ```yml
+# Name of this GitHub Actions workflow.
 name: Semgrep
 
 on:
+  # Scan changed files in PRs (diff-aware scanning):
   pull_request: {}
+  # Scan on-demand through GitHub Actions interface:
   workflow_dispatch: {}
+  # Scan mainline branches and report all findings:
   push:
     branches: ["master", "main"]
+  # Schedule the CI job (this method uses cron syntax):
   schedule:
     - cron: '20 17 * * *' # Sets Semgrep to scan every day at 17:20 UTC.
-    
-permissions: 
-    security-events: write
+    # It is recommended to change the schedule to a random time.
+
 jobs:
   semgrep:
+    # User definable name of this GitHub Actions job.
     name: semgrep/ci 
+    # If you are self-hosting, change the following `runs-on` value: 
     runs-on: ubuntu-latest
 
     container:
       # A Docker image with Semgrep installed. Do not change this.
       image: returntocorp/semgrep
 
+    # Skip any PR created by dependabot to avoid permission issues:
     if: (github.actor != 'dependabot[bot]')
-
+    permissions:
+      # required for all workflows
+      security-events: write
+      # for workflows in private repos
+      actions: read
+      contents: read
     steps:
       - uses: actions/checkout@v3
-      - run: semgrep ci --sarif > semgrep.sarif
-        env:
-          SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
```

Comment:
I don't see this elsewhere in the file, so I don't think this line should be deleted. Pretty important to have the token. :)